crates/red_knot_python_semantic/resources/mdtest/directives/assert_type.md

```diff
@@ -0,0 +1,123 @@
+from typing import reveal_type
+
+# `assert_type`
```

Comment:
One thing that we should probably test here somewhere (if only to document our behavior) is the handling of `Literal` types. And whether we use the inferred type or the declared type. For example, pyright behaves like this:
```py
from typing import assert_type

x: int = 1

assert_type(x, int)  # "assert_type" mismatch: expected "int" but received "Literal[1]"

def f():
    assert_type(x, int)  # fine (uses declared type)
```

It looks like we would currently have the same behavior?